compiler/src/test/java/motif/compiler/TestHarness.java

```diff
@@ -29,45 +30,62 @@
 import org.junit.runner.RunWith;
 import org.junit.runners.Parameterized;
 
-import javax.tools.Diagnostic;
-import javax.tools.JavaFileObject;
-import java.io.BufferedWriter;
-import java.io.File;
-import java.io.FileWriter;
-import java.io.IOException;
+import javax.annotation.Nullable;
+import java.io.*;
 import java.lang.reflect.InvocationTargetException;
-import java.net.MalformedURLException;
 import java.net.URL;
 import java.net.URLClassLoader;
 import java.nio.charset.Charset;
-import java.util.Arrays;
-import java.util.Collection;
-import java.util.List;
-import java.util.Locale;
+import java.nio.file.Path;
+import java.util.*;
+import java.util.function.Predicate;
 import java.util.stream.Collectors;
 
-import static com.google.testing.compile.CompilationSubject.assertThat;
+import static com.google.common.truth.Truth.assertWithMessage;
 
 @RunWith(Parameterized.class)
 public class TestHarness {
```

Comment:
General question: Why is this class in Java, but most of the project is in Kotlin?

Reply:
This is for legacy reasons. We can convert to kotlin in the future.